galley/pkg/config/util/kuberesource/resources.go

```diff
@@ -60,11 +81,27 @@ func isKindExcluded(excludedResourceKinds []string, kind string) bool {
 	return false
 }
 
-// DefaultExcludedResourceKinds returns the default list of resource kinds to exclude, which is the builtin types.
-func DefaultExcludedResourceKinds() []string {
-	resources := make([]string, 0)
-	for _, spec := range builtin.GetSchema().All() {
-		resources = append(resources, spec.Kind)
+func getUpstreamCollections(inputs collection.Names, xformProviders transformer.Providers) map[collection.Name]struct{} {
```

Comment:
Should this be defined in the `provider` package on `Provider` or `Providers`? Something very similar is also defined in galley/pkg/config/analysis/local/analyze.go. Maybe something like 
`providers.RequiredInputsFor(outputs)`?

Reply:
What's in analyze.go is being moved here. 

I agree that it makes sense to define this on providers.